test/v1alpha1/service.go

```diff
@@ -143,6 +144,43 @@ func CreateRunLatestServiceReady(t *testing.T, clients *test.Clients, names *tes
 	return resources, err
 }
 
+// CreateRunLatestServiceNotReady creates a new Service expecting that it will not be ready. This function expects Service and Image name passed in through 'names'.
+// Names is updated with the Route and Configuration created by the Service and ResourceObjects is returned with the Service, Route, and Configuration objects.
+// Returns the error if not able to create a service that's not ready. If service comes up correctly, an error is returned.
+func CreateRunLatestServiceNotReady(t *testing.T, clients *test.Clients, names *test.ResourceNames, fopt ...rtesting.ServiceOption) error {
+	if names.Image == "" {
+		return fmt.Errorf("expected non-empty Image name; got Image=%v", names.Image)
+	}
+
+	t.Logf("Creating a new Service %s.", names.Service)
+	svc, err := CreateLatestService(t, clients, *names, fopt...)
```

Comment:
You should just use this directly in your test. I don't think the rest of this function is really necessary to accomplish what you want.